feat: reintroduce private sw-dashboard-statistics

MalteJanz opened the pull request:
Reintroduces the `sw-dashboard-statistics` component that was originally removed and remains deprecated in `6.6.x`. Some remarkable changes:
- All extension points like twig blocks or position-identifiers were removed
- The implementation is marked `@private` and might be removed at any time (even in a minor update!), this ensures that we are able to replace it with shopware analytics without waiting for a major shopware version.

![image](https://github.com/user-attachments/assets/91c442a6-c476-4e7f-aa21-749dd3b42698)


closes [NEXT-40766](https://shopware.atlassian.net/browse/NEXT-40766)


[NEXT-40766]: https://shopware.atlassian.net/browse/NEXT-40766?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

MalteJanz committed:
feat: reintroduce private sw-dashboard-statistics

closes NEXT-40766

MalteJanz committed:
fix tests and use mt-card

fschmtt reviewed (approved):
Just to make everyone aware once again: We're still removing the previously existing extension points (which is IMO covered by the deprecation).

There are alternative extension points before and after the dashboard content.

@seggewiss do you see any issue with that?

mitelg commented:
can we make sure, that the PHP parts will be removed, once they are not needed anymore?

fschmtt commented:
@mitelg 
> can we make sure, that the PHP parts will be removed, once they are not needed anymore?

surely agree, but could you please reify what concretely you're referring to? there were no related PHP part marked as deprecated until now 🙈

MalteJanz commented:
> can we make sure, that the PHP parts will be removed, once they are not needed anymore?

@mitelg As far as I can see the `DashboardController` and `OrderAmountService` are already declared as `@internal`. But I'm not sure if we can also remove that controller HTTP endpoint with a minor, at least it doesn't show up on our [API Docs](https://shopware.stoplight.io/docs/admin-api) 🤔

MalteJanz committed:
fix meta.spec.js to ignore empty or null position-identifiers

diff --git c/src/Administration/Resources/app/administration/src/meta/meta.spec.js i/src/Administration/Resources/app/administration/src/meta/meta.spec.js
index 7fb230f708..26968390c2 100644
--- c/src/Administration/Resources/app/administration/src/meta/meta.spec.js
+++ i/src/Administration/Resources/app/administration/src/meta/meta.spec.js
@@ -122,7 +122,9 @@ describe('Administration meta tests', () => {
                 [...fileContent.matchAll(/position-identifier="(.*)"/gm)]
                     .map((match) => match[1])
                     .forEach((match) => {
-                        result.push(match);
+                        if (match !== '' && match !== 'null') {
+                            result.push(match);
+                        }
                     });
             });

MalteJanz commented on src/Administration/Resources/app/administration/src/meta/meta.spec.js:
```diff
@@ -122,7 +122,9 @@ describe('Administration meta tests', () => {
                     .map((match) => match[1])
                     .forEach((match) => {
-                        result.push(match);
+                        if (match !== '' && match !== 'null') {
+                            result.push(match);
```
@seggewiss Fyi: I adjusted the `meta.spec.js` test to not complain about added empty `position-identifier` attributes and removed the two empty strings from `src/Administration/Resources/app/administration/src/meta/position-identifiers.json`

larskemper replied:
i will close framework/admin have approved

mitelg commented:
> @mitelg As far as I can see the `DashboardController` and `OrderAmountService` are already declared as `@internal`. But I'm not sure if we can also remove that controller HTTP endpoint with a minor, at least it doesn't show up on our [API Docs](https://shopware.stoplight.io/docs/admin-api) 🤔

ah thanks. that should be sufficient then :+1: people also should not rely on an API, that is marked as internal. so it would be safe to delete that, too

larskemper committed:
fix: regenerate positon-indentifiers
